src/js/popup.js

```diff
@@ -1454,6 +1454,35 @@ Logic.registerPanel(P_CONTAINER_ASSIGNMENTS, {
     return Promise.resolve(null);
   },
 
+  async addAssignment() {
+    const errorMessage = document.getElementById("edit-sites-add-assignment-error");
+    const assignmentInput = document.getElementById("edit-sites-add-assignment-info");
+
+    let url = document.getElementById("edit-sites-add-assignment-info").value;
+
+    if (!URL.canParse(url)) {
+      url = "http://" + url;
+    }
+
+    if (new URL(url).protocol == "https:" || new URL(url).protocol == "http:") {
+      url = new URL(url);
+    } else {
+      errorMessage.style.display = "block";
+      assignmentInput.style.border = "2px solid var(--input-destructive-border-color)";
+    }
+
+    const userContextId = Logic.currentUserContextId();
+    await Utils.setOrRemoveAssignment(
+      false, url.origin, userContextId, false
```

Comment:
`url.origin` can be non-existant/`undefined` due to the check above...

Reply:
What "check" are you referring to?

Reply:
```javascript
   if (new URL(url).protocol == "https:" || new URL(url).protocol == "http:") {
      url = new URL(url);
   } else {
```
As you can see you change `url` here to be an `URL()`, but before this check `url` was a `string`